source/operations/concepts/erasure-coding.rst

```diff
@@ -10,62 +10,131 @@ Erasure Coding
+
+   :math:`N (ERASURE\ SET\ SIZE) = K (DATA) + M (PARITY)`
+
+   .. figure:: /images/erasure/erasure-coding-shard-ec6.svg
+      :figwidth: 100%
+      :align: center
+      :alt: Diagram of an object written with EC:6 to an erasure set with 16 drives
+
+   This deployment has an erasure set size of ``16`` and a parity of ``EC:6``. 
+   Based on the formula, MinIO creates ``10`` data shards and ``6`` parity shards for each object.
+
+MinIO can use any parity shard to reconstruct any data shard.
+   .. figure:: /images/erasure/erasure-coding-shard-reconstruction.svg
+      :figwidth: 100%
+      :align: center
+      :alt: Diagram of MinIO using parity shards to reconstruct data shards
+
+      An object has lost two data shards.
+      MinIO uses any two of the available parity shards to reconstruct the data shards.
+      Even a single remaining parity shard would be sufficient to reconstruct both missing data shards.
+
+MinIO requires a minimum of ``K`` Data *or* Parity shards to **read** the object.
+   The value ``K`` constitutes the **read quorum** for the deployment.
+   Alternatively, the object can tolerate the loss of no more than ``M`` shards of any time while remaining readable.
+
+   .. figure:: /images/erasure/erasure-coding-shard-read-quorum.svg
+      :figwidth: 100%
+      :align: center
+      :alt: Diagram of an object maintaining read quorum with only 10 total shards remaining
+
+      An object has 10 data shards and 6 parity shards for a total of 16 shards.
+      It has lost 3 data and 3 parity shards for a total of 6 shards lost and 10 remaining.
+      The loss of any further shards would result in losing read quorum.
+
+   An object that has lost read quorum is effectively "lost" and may be recovered through other means such as :ref:`replication resynchronization <minio-bucket-replication-resynchronize>`.
+
+MinIO similarly requires successful creation of at least ``K`` data shards to **write** the object.
+   The value ``K`` constitutes the **write quorum** for the deployment.
+   A deployment could lose write quorum due to having insufficient online drives, insufficient drive space, or network issues between nodes.
+
+   .. figure:: /images/erasure/erasure-coding-shard-write-quorum.svg
+      :figwidth: 100%
+      :align: center
+      :alt: Diagram of an object maintaining write quorum with only 10 shards remaining.
+
+      MinIO creates 16 shards for an object with parity of ``EC:6``.
+      Due to drive errors, only 10 drives remain online for MinIO to use for write operations.
+      This meets write quorum, but the loss of any further drives would result in write failure for newer objects.
+
+   If Parity (``EC:M``) is exactly 1/2 the number of drives in the erasure set, **write quorum** is ``K+1``.
+   This extra shard prevents data inconsistency due to "split-brain"-types of failure states
 
 .. _minio-ec-erasure-set:
 
-Erasure Sets
-------------
+Erasure Set and Distribution
```

Comment:
Consider having this section at the top before the discussion of data and parity shards.
It seems to me that understanding an erasure set is a precursor to understanding how sharding works.